src/recon_test/test_view_offset_root.cxx

```diff
@@ -0,0 +1,242 @@
+    {
+      // only stores prompts
+        if (record.is_event() && record.event().is_prompt())
+        {
+          Bin bin;
+          bin.set_bin_value(1.f);
+          // gets the bin corresponding to the event
+          record.event().get_bin(bin, *lm_data_sptr->get_proj_data_info_sptr());
+          if ( bin.get_bin_value()>0 )
+          {
+            // computes the TOF probabilities along the bin LOR
+            proj_matrix_sptr->get_proj_matrix_elems_for_one_bin(proj_matrix_row, bin);
+            // adds coordinates and weights of the elements with highest probability along LOR
+            if (!test_LOR_closest_approach(proj_matrix_row, test_discretised_density_sptr,
+                                           original_coords, tolerance))
+            {
+              failed_events += 1;
+            }
+            tested_events += 1;
+          }
+        }
+      }
+    }
+
+  cerr << "\nNUMBER OF FAILED EVENTS = " << failed_events << "\t NUMBER OF TESTED EVENTS = " << tested_events << std::endl;
+  check_if_less(failed_events, 0.5 * tested_events,
+                "the number of failed events is more than half the number of tested events.");
+
+}
+
+bool
+ROOTconsistency_Tests::
+test_LOR_closest_approach(const ProjMatrixElemsForOneBin &probabilities,
+                          const shared_ptr <DiscretisedDensity<3, float>> &test_discretised_density_sptr,
+                          const CartesianCoordinate3D<float> original_coords,
+                          const float tolerance)
+{
+  // Loop variables
+  CartesianCoordinate3D<float> closest_LOR_voxel_to_origin;
+  float min_distance;
+  bool first_entry = true; // Use this to populate with the initial value
+
+  ProjMatrixElemsForOneBin::const_iterator element_ptr = probabilities.begin();
+  // iterate over all to element_ptr to find the minimal distance between LOR and original_coords
+  while (element_ptr != probabilities.end())
+  {
+    CartesianCoordinate3D<float> voxel_coords =
+            test_discretised_density_sptr->get_physical_coordinates_for_indices(element_ptr->get_coords());
+
+    float dist_to_original = norm(voxel_coords - original_coords);
+    if (dist_to_original < min_distance || first_entry)
+    {
+      closest_LOR_voxel_to_origin = voxel_coords;
+      min_distance = dist_to_original;
+      first_entry = false;
+    }
+    ++element_ptr;
+  }
+//  if (!check_if_less(min_distance, 3*norm(grid_spacing), "ERR msgs"))
+  if (min_distance > tolerance)
+  {
+//    cerr << "min_distance = " << min_distance << "\t tolerance = " << tolerance << std::endl;
+    return false; // Test failed - LOR closest voxel beyond tolerance
+  }
+  return true; // Test passed - LOR closest voxel within tolerance
+}
+
+END_NAMESPACE_STIR
+
+int main(int argc, char **argv)
+{
+  USING_NAMESPACE_STIR
+
+  if (argc != 3)
+  {
+    cerr << "Usage : hroot_filename generate_image_par_filename\n\n";
+    return EXIT_FAILURE;
+  }
+
+
+  ifstream in(argv[1]);
```

Comment:
put all these checks in its own block `{ ... }` such that the files get closed before passing it on. Doesn't matter, but cleaner